### PR TITLE
[TTAHUB-496] Update target populations on legacy reports

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change
Create script to update legacy report target populations
Specifically
- Infant/Toddlers to Infants and Toddlers (ages birth to 3)
- Preschool to Preschool (ages 3-5)
- Affected by Homelessness to Children Experiencing Homelessness

This will remove duplicated values in analytics widgets like on the recipient record page

## How to test
- (Locally) Verify that you have legacy data that needs updating. Note some reports that you can check after running the script. 
- Run ```yarn updateLegacyPopulations:local```. 
- Check those reports you noted earlier.
- Run the script again. It should find no reports to update.
- All tests pass.

### Successful run on sandbox
Screen Shot 2021-12-02 at 9 46 16 AM

You can also compare the recipient Absaroka Head Start - Region 8 on sandbox and dev to further verify success

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-496

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [n/a] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [n/a] API Documentation updated
- [n/a] Boundary diagram updated
- [n/a] Logical Data Model updated
- [n/a] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: src/tools/updateLegacyPopulations.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import { Op } from 'sequelize';
+import { ActivityReport } from '../models';
+import { auditLogger } from '../logger';
+
+const newPopulations = {
+  'Infant/Toddlers': 'Infants and Toddlers (ages birth to 3)',
+  Preschool: 'Preschool (ages 3-5)',
```
Comment (GarrettEHill) [inline src/tools/updateLegacyPopulations.js:7]: should Preschool be in quotes?

Comment (thewatermethod) [reply src/tools/updateLegacyPopulations.js:7]: Doesn't need to be (also, the linter won't allow it)

**Review by kryswisnaskas — CHANGES_REQUESTED** [review]
This looks good as far as changing the values in the AR table, the `targetPopulations ` column. 👍 
 Can we also make sure the UI reflects the changes? I believe the easiest way would be to also modify the values in the `imported` column.

This can be checked, for example on report R11-AR-000214

image

**Comment by thewatermethod** [comment]
> This looks good as far as changing the values in the AR table, the `targetPopulations ` column. 👍 Can we also make sure the UI reflects the changes? I believe the easiest way would be to also modify the values in the `imported` column.
> 

Good call - I'll do that now

**Review by GarrettEHill — APPROVED** [review]

**Comment by kryswisnaskas** [comment]
👍

**Review by kryswisnaskas — APPROVED** [review]

File: src/tools/updateLegacyPopulations.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  auditLogger.info(`Updating legacy target population data...
+  
+  `);
+
+  const reports = await ActivityReport.findAll({
+    attributes: ['id', 'targetPopulations', 'imported'],
+    where: {
+      [Op.or]: [
+        {
+          targetPopulations: {
```
Comment (jasalisbury) [inline src/tools/updateLegacyPopulations.js:21]: Is this also updating non-legacy reports?

Comment (thewatermethod) [reply src/tools/updateLegacyPopulations.js:21]: Should be both, yeah

**Review by jasalisbury — APPROVED** [review]

**Pull request merged**